File: common/src/main/java/dev/chasem/cobblemonextras/commands/PokeShout.java
package dev.chasem.cobblemonextras.commands;

import com.cobblemon.mod.common.Cobblemon;
import com.cobblemon.mod.common.api.pokemon.stats.Stats;
import com.cobblemon.mod.common.api.storage.party.PlayerPartyStore;
import com.cobblemon.mod.common.pokemon.Pokemon;
import com.cobblemon.mod.common.util.LocalizationUtilsKt;
import com.mojang.brigadier.CommandDispatcher;
import com.mojang.brigadier.arguments.IntegerArgumentType;
import com.mojang.brigadier.context.CommandContext;
import dev.chasem.cobblemonextras.CobblemonExtras;
import dev.chasem.cobblemonextras.permissions.CobblemonExtrasPermissions;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.text.*;
import net.minecraft.util.Formatting;

import java.util.List;

import static dev.chasem.cobblemonextras.util.PokemonUtility.getHoverText;
import static net.minecraft.server.command.CommandManager.argument;
import static net.minecraft.server.command.CommandManager.literal;

public class PokeShout {

    public void register(CommandDispatcher<ServerCommandSource> dispatcher) {
        dispatcher.register(
                literal("pokeshout")
                        .requires(src -> CobblemonExtrasPermissions.checkPermission(src, CobblemonExtras.permissions.POKESHOUT_PERMISSION))
                        .then(argument("slot", IntegerArgumentType.integer(1, 6)).executes(this::execute))
        );
    }

    private int execute(CommandContext<ServerCommandSource> ctx) {
        if (ctx.getSource().getPlayer() != null) {
            ServerPlayerEntity player = ctx.getSource().getPlayer();
            Integer slot = ctx.getArgument("slot", Integer.class);
            PlayerPartyStore party = Cobblemon.INSTANCE.getStorage().getParty(player);
            Pokemon pokemon = party.get(slot - 1);
            if (pokemon != null) {
                MutableText toSend = Text.literal("[").formatted(Formatting.GREEN)
                        .append(Text.literal("PokeShout").formatted(Formatting.YELLOW))
                        .append(Text.literal("] ").formatted(Formatting.GREEN))
                        .append(player.getDisplayName().copy().append(Text.of(": ")).formatted(Formatting.WHITE));
                MutableText pokemonName = pokemon.getSpecies().getTranslatedName().formatted(Formatting.GREEN).append(" ");
                toSend.append(pokemonName);
                if (pokemon.getShiny()) {
                    toSend.append(Text.literal("★ ").formatted(Formatting.GOLD));
                }
                getHoverText(toSend, pokemon);
                ctx.getSource().getServer().getPlayerManager().getPlayerList().forEach(serverPlayer -> serverPlayer.sendMessage(toSend));
            } else {
                ctx.getSource().sendError(Text.literal("No Pokemon in slot."));
            }
        } else {
            ctx.getSource().sendError(Text.of("Sorry, this is only for players."));
        }
        return 1;
    }



}

File: common/src/main/kotlin/dev/chasem/cobblemonextras/commands/PokeShout.kt
package dev.chasem.cobblemonextras.commands

import com.cobblemon.mod.common.Cobblemon
import dev.chasem.cobblemonextras.CobblemonExtras
import dev.chasem.cobblemonextras.permissions.CobblemonExtrasPermissions
import dev.chasem.cobblemonextras.util.PokemonUtility
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands
import net.minecraft.server.level.ServerPlayer
import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.arguments.IntegerArgumentType
import com.mojang.brigadier.context.CommandContext
import net.minecraft.ChatFormatting
import net.minecraft.network.chat.Component

class PokeShout {
    fun register(dispatcher: CommandDispatcher<CommandSourceStack>) {
        dispatcher.register(
                Commands.literal("pokeshout")
                        .requires { src: CommandSourceStack? -> CobblemonExtrasPermissions.checkPermission(src, CobblemonExtras.permissions.POKESHOUT_PERMISSION) }
                        .then(Commands.argument("slot", IntegerArgumentType.integer(1, 6)).executes { ctx: CommandContext<CommandSourceStack> -> this.execute(ctx) })
        )
    }

    private fun execute(ctx: CommandContext<CommandSourceStack>): Int {
        if (ctx.source.player != null) {
            val player: ServerPlayer = ctx.source.player!!
            val slot: Int = ctx.getArgument<Int>("slot", Int::class.java)
            val party = Cobblemon.storage.getParty(player)
            val pokemon = party.get(slot - 1)
            if (pokemon != null) {
                val toSend = Component.literal("[").withStyle(ChatFormatting.GREEN)
                        .append(Component.literal("PokeShout").withStyle(ChatFormatting.YELLOW))
                        .append(Component.literal("] ").withStyle(ChatFormatting.GREEN))
                        .append(player.displayName!!.copy().append(Component.literal(": ")).withStyle(ChatFormatting.WHITE))
                val pokemonName = pokemon.species.translatedName.withStyle(ChatFormatting.GREEN).append(" ")
                toSend.append(pokemonName)
                if (pokemon.shiny) {
                    toSend.append(Component.literal("★ ").withStyle(ChatFormatting.GOLD))
                }
                PokemonUtility.getHoverText(toSend, pokemon)
                ctx.source.server.playerList.players.forEach { serverPlayer -> serverPlayer.sendSystemMessage(toSend) }
            } else {
                ctx.source.sendFailure(Component.literal("No Pokemon in slot."))
            }
        } else {
            ctx.source.sendFailure(Component.literal("Sorry, this is only for players."))
        }
        return 1
    }
}
